src/modules/wallet/sagas.spec.ts

```diff
@@ -1,25 +1,113 @@
 import { expectSaga } from 'redux-saga-test-plan'
 import * as matchers from 'redux-saga-test-plan/matchers'
-import { call } from 'redux-saga/effects'
+import { call, race, take } from 'redux-saga/effects'
 import { ChainId } from '@dcl/schemas/dist/dapps/chain-id'
 import { Provider } from 'decentraland-connect'
-import { createWalletSaga } from './sagas'
+import { createWalletSaga, getAccount } from './sagas'
 import {
+  FETCH_WALLET_FAILURE,
+  FETCH_WALLET_SUCCESS,
+  connectWalletFailure,
+  connectWalletRequest,
+  connectWalletSuccess,
+  enableWalletFailure,
+  enableWalletRequest,
+  enableWalletSuccess,
+  fetchWalletRequest,
   switchNetworkFailure,
   switchNetworkRequest,
   switchNetworkSuccess
 } from './actions'
 import { getConnectedProvider, getNetworkProvider } from '../../lib/eth'
 import { switchProviderChainId } from './utils'
+import { ProviderType, Wallet } from './types'
 
 const walletSaga = createWalletSaga({ CHAIN_ID: 1 })
+const error = 'anErrorMessage'
 
 describe('Wallet sagas', () => {
   let mockProvider: Provider
+
   beforeAll(async () => {
     mockProvider = await getNetworkProvider(ChainId.ETHEREUM_MAINNET)
   })
 
+  describe('when handling the action to connect the wallet', () => {
+    describe('and the fetch wallet request fails', () => {
+      it('should dispatch the fetch wallet request, and an action to signal the failure', () => {
+        return expectSaga(walletSaga)
+          .provide([
+            [
+              race({
+                success: take(FETCH_WALLET_SUCCESS),
+                failure: take(FETCH_WALLET_FAILURE)
+              }),
+              { success: null, failure: { payload: { error } } }
+            ]
+          ])
```

Comment:
What if we mock, in the provide section, the `put(fetchWalletRequest())` so the action doesn't get handled?
The same could be applied to the other tests of the saga that put the same action.